### PR TITLE
Add arm64 review builds

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

The review builds fail if `macos-latest` resolves to `macos-14` (`osx-amr64`).

See https://github.com/conda/conda/actions/runs/8809949546/job/24181531935?pr=13837

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [ ] ~Add / update necessary tests?~
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

**Comment by kenodegard** [comment]
Oh this depends on https://github.com/conda/actions/pull/172 being merged

**Comment by jaimergp** [comment]
See https://github.com/conda/conda/pull/13836

File: .github/workflows/builds-review.yaml [hunk 1]
```diff
@@ -53,7 +55,7 @@ jobs:
           fetch-depth: 0
 
       - name: Create and upload review build
-        uses: conda/actions/canary-release@f46142e89fa703cc69f0421ca1d313ab2d5bfff6
+        uses: conda/actions/canary-release@miniconda-latest
```
Comment (kenodegard) [inline .github/workflows/builds-review.yaml:58]:
```suggestion
        uses: conda/actions/canary-release@f46142e89fa703cc69f0421ca1d313ab2d5bfff6
```

Comment (kenodegard) [inline .github/workflows/builds-review.yaml:58]:
```suggestion
        uses: conda/actions/canary-release@f46142e89fa703cc69f0421ca1d313ab2d5bfff6
```

**Review by beeankha — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**